File: Java/src/test/java/com/gildedrose/TexttestFixture.java
package com.gildedrose;

public class TexttestFixture {
    public String golden_master(String[] args) {
        System.out.println("OMGHAI!");

        Item[] items = new Item[]{
            new Item("+5 Dexterity Vest", 10, 20), //
            new Item("Aged Brie", 2, 0), //
            new Item("Elixir of the Mongoose", 5, 7), //
            new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
            new Item("Sulfuras, Hand of Ragnaros", -1, 80),
            new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
            new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
            // this conjured item does not work properly yet
            new Item("Conjured Mana Cake", 3, 6)};

        StringBuilder string_to_return = new StringBuilder();

        GildedRose app = new GildedRose(items);

        int days = 2;
        if (args.length > 0) {
            days = Integer.parseInt(args[0]) + 1;
        }

        for (int i = 0; i < days; i++) {
            string_to_return.append("-------- day ").append(i).append(" --------\n");
            string_to_return.append("name, sellIn, quality");
            for (Item item : items) {
                string_to_return.append(item);
            }
            string_to_return.append("\n");
            app.updateQuality();
        }
        string_to_return.toString();
        return string_to_return;
    }
}

File: python/texttest_fixture.py
# -*- coding: utf-8 -*-
from __future__ import print_function
from gilded_rose import *
from item_classes import *


def golden_master_print(days):
    print("OMGHAI!")
    items = [
             Basic(name="+5 Dexterity Vest", sell_in=10, quality=20),
             Brie(name="Aged Brie", sell_in=2, quality=0),
             Basic(name="Elixir of the Mongoose", sell_in=5, quality=7),
             Legendary(name="Sulfuras, Hand of Ragnaros", sell_in=0, quality=80),
             Legendary(name="Sulfuras, Hand of Ragnaros", sell_in=-1, quality=80),
             Backstage(name="Backstage passes to a TAFKAL80ETC concert", sell_in=15, quality=20),
             Backstage(name="Backstage passes to a TAFKAL80ETC concert", sell_in=10, quality=49),
             Backstage(name="Backstage passes to a TAFKAL80ETC concert", sell_in=5, quality=49),
             Basic(name="Conjured Mana Cake", sell_in=3, quality=6),  # <-- :O
            ]

    import sys
    if len(sys.argv) > 1:
        days = int(sys.argv[1]) + 1
    for day in range(days):
        print("-------- day %s --------" % day)
        print("name, sellIn, quality")
        for item in items:
            print(item)
        print("")
        GildedRose(items).update_quality()


golden_master_print(31)
